File: frontend/src/types/patent.ts
export interface Patent {
  id: string;
  title: string;
  summary: string;
  applicant: string;
  inventor: string;
  applicationDate: string;
  documentDate: string;
  mainIpc: string;
  ipc: string;
  legalStatus: string;
  type: string;
  documentNumber: string;
  relevance_score?: number;
  source_innovation?: string;
  search_query?: string;

  // 兼容旧字段
  abstract?: string;
  applicants?: string[];
  inventors?: string[];
  filingDate?: string;
  publicationDate?: string;
  patentNumber?: string;
  ipcCodes?: string[];
}

export interface PatentSearchQuery {
  q: string;
  page?: number;
  size?: number;
}

File: backend/app/models/patent.py
from pydantic import BaseModel
from typing import Optional


class Patent(BaseModel):
    id: str
    title: str
    abstract: str
    applicants: list[str]
    inventors: list[str]
    filing_date: str
    publication_date: str
    patent_number: str
    ipc_codes: list[str]
    relevance_score: int


class PatentStats(BaseModel):
    total_count: int
    related_count: int
    core_count: int
    analyzed_count: int
    top_patents: list[Patent]
